### ISSUE TITLE
CMake build question

### ISSUE THREAD

**Issue by Luonic** [halide/Halide#4775]
I am trying to set up project structure around CMake to build simple image processing app
I would like to be able to use filter libs from root of project and generate libs via CMakeLists.txt in generators directory.
Project hierarchy is simple:
```
root
- CMakeLists.txt (1)
- main.cpp
- generators
  - CMakeLists.txt (2)
  - adaptive_contrast.cpp
  - sRGB_to_linearRGB.cpp
```

CMakeLists.txt (1):
```
cmake_minimum_required(VERSION 3.13)
project(image-processing VERSION 0.1.0)

set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED on)
# set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror")

include(CTest)
enable_testing()

add_subdirectory(generators)
# include(${CMAKE_CURRENT_SOURCE_DIR}/generators/CMakeLists.txt)

add_executable(image-processing main.cpp)

target_include_directories(image-processing PUBLIC ~/Code/Halide/include ~/Code/Halide/tools)

find_library(HALIDE_LIBRARY Halide ~/Code/Halide/bin)
find_library(PTHREAD_LIBRARY pthread)
find_library(JPEG_LIBRARY jpeg)
find_library(PNG_LIBRARY png)

target_link_libraries(image-processing PUBLIC adaptive_contrast)
target_link_libraries(image-processing PUBLIC srgb_to_linearrgb)
target_link_libraries(image-processing PUBLIC ${HALIDE_LIBRARY} ${PTHREAD_LIBRARY} ${JPEG_LIBRARY} ${PNG_LIBRARY} ${CMAKE_DL_LIBS})


set(CPACK_PROJECT_NAME ${PROJECT_NAME})
set(CPACK_PROJECT_VERSION ${PROJECT_VERSION})
include(CPack)
```

CMakeLists.txt (2):
```
set(HALIDE_DISTRIB_DIR "/home/alex/Code/Halide")
include("${HALIDE_DISTRIB_DIR}/halide_config.cmake")
include("${HALIDE_DISTRIB_DIR}/halide.cmake")

# Adaptive contrast
halide_library(adaptive_contrast SRCS adaptive_contrast.cpp)

# sRGB to Linear RGB
halide_library(srgb_to_linearrgb SRCS sRGB_to_linearRGB.cpp)
```

When i build any target in my project i get:
```
[main] Building folder: image-processing image-processing
[build] Starting build
[proc] Executing command: /usr/local/bin/cmake --build /home/alex/Code/image-processing/build --config Debug --target image-processing -- -j 18
[build] [1/2  50% :: 1.939] Building CXX object CMakeFiles/image-processing.dir/main.cpp.o
[build] FAILED: CMakeFiles/image-processing.dir/main.cpp.o 
[build] /bin/clang++-9   -I/home/alex/Code/Halide/include -I/home/alex/Code/Halide/tools -I../generators -g   -std=gnu++17 -MD -MT CMakeFiles/image-processing.dir/main.cpp.o -MF CMakeFiles/image-processing.dir/main.cpp.o.d -o CMakeFiles/image-processing.dir/main.cpp.o -c ../main.cpp
[build] ../main.cpp:16:10: fatal error: 'generators/adaptive_contrast.hpp' file not found
[build] #include <generators/adaptive_contrast.hpp>
[build]          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[build] 1 error generated.
[build] ninja: build stopped: subcommand failed.
[build] Build finished with exit code 1
```

I have not found any example of using  cmake with subdirectories and including them. Could someone help me?

**Comment by alexreinking** [comment]
I'll circle back to this once #4644 lands. The Halide+CMake story is about to change pretty significantly.

**Comment by alexreinking** [comment]
Please read the new `README_cmake.md` and open a new issue if you still need help.
